src/resources/functions.ts

```diff
@@ -1,41 +1,76 @@
-import { ServerlessFunction, Serverless } from '../serverlessPlugin';
+import { Serverless, ServerlessFunction } from "../serverlessPlugin"
 
 export default (serverless: Serverless) => {
-    const handlerPath = 'swagger/';
-    const configInput = serverless?.configurationInput || serverless.service;
-    const path = serverless.service.custom?.autoswagger?.swaggerPath ?? 'swagger';
-    const name =
-        typeof configInput?.service == 'object' ? configInput.service.name : configInput.service;
-    const stage = configInput?.provider?.stage;
+  const handlerPath = "swagger/"
+  const configInput = serverless?.configurationInput || serverless.service
+  const path = serverless.service.custom?.autoswagger?.swaggerPath ?? "swagger"
+  const name =
+    typeof configInput?.service == "object"
+      ? configInput.service.name
+      : configInput.service
+  const stage = configInput?.provider?.stage
+
+  const useStage = serverless.service.custom?.autoswagger?.useStage
+
+  if (useStage) {
+    return {
+      swaggerUI: {
+        name: name && stage ? `${name}-${stage}-swagger-ui` : undefined,
+        handler: handlerPath + "swaggerhtml.handler",
+        disableLogs: true,
+        events: [
+          {
+            httpApi: {
+              method: "get",
+              path: `/${stage}/swagger`,
```

Comment:
path: useStage ? `/${stage}/swagger` : `/swagger`,

that would prevent code duplication in this big if else

Reply:
Ooh, thanks for the tip :)

I was doing that on another way and when i changed it, i forgot a code review hahaha

I think there's another places that we can improve too, as JS/TS is not a language that i am used to, i think that a really have a lot to improve 

Maybe when Sam has some time he could improve my code :)

Reply:
@desaintvincent i submited another commit with your correction, thanks :)